src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -684,6 +691,20 @@ public function prependExtensionConfig(string $name, array $config)
         array_unshift($this->extensionConfigs[$name], $config);
     }
 
+    /**
+     * Adds a parameter available during at compile-time only, it will be removed when dumped.
+     */
+    public function setBuildParameter(string $parameter, array|bool|string|int|float|\UnitEnum|null $value): void
+    {
+        $this->parameterBag->set($parameter, $value);
+        $this->buildParameters[$parameter] = false;
+    }
+
+    public function removeParameter(string $parameter): void
+    {
+        $this->parameterBag->remove($parameter);
```

Comment:
no need for the removeParamter method I suppose

Reply:
You're right, for DX purpose this could be a dedicated PR if needed.